### ISSUE TITLE
Java 11 support

### ISSUE THREAD

**Issue by gfreezy** [apache/incubator-shardingsphere#3553]
I can  run sharding-jdbc under java 11 after adding some dependencies.

```
javax.xml.bind:jaxb-api:2.3.0
javax.activation:activation:1.1
org.glassfish.jaxb:jaxb-runtime:2.3.0
```

Shall i send a PR?

```java
Exception in thread "main" java.lang.NoClassDefFoundError: javax/xml/bind/JAXBContext
	at org.apache.shardingsphere.core.parse.rule.jaxb.loader.extractor.ExtractorRuleDefinitionEntityLoader.load(ExtractorRuleDefinitionEntityLoader.java:39)
	at org.apache.shardingsphere.core.parse.rule.registry.ParseRuleRegistry.initParseRuleDefinition(ParseRuleRegistry.java:62)
	at org.apache.shardingsphere.core.parse.rule.registry.ParseRuleRegistry.<init>(ParseRuleRegistry.java:58)
	at org.apache.shardingsphere.core.parse.rule.registry.ShardingParseRuleRegistry.<init>(ShardingParseRuleRegistry.java:28)
	at org.apache.shardingsphere.core.parse.rule.registry.ShardingParseRuleRegistry.<clinit>(ShardingParseRuleRegistry.java:31)
	at org.apache.shardingsphere.core.parse.entry.ShardingSQLParseEntry.getSQLParseEngine(ShardingSQLParseEntry.java:45)
	at org.apache.shardingsphere.core.parse.entry.SQLParseEntry.parse(SQLParseEntry.java:48)
	at org.apache.shardingsphere.core.route.router.sharding.ParsingSQLRouter.parse(ParsingSQLRouter.java:76)
	at org.apache.shardingsphere.core.route.PreparedStatementRoutingEngine.route(PreparedStatementRoutingEngine.java:64)
	at org.apache.shardingsphere.core.PreparedQueryShardingEngine.route(PreparedQueryShardingEngine.java:60)
	at org.apache.shardingsphere.core.BaseShardingEngine.executeRoute(BaseShardingEngine.java:86)
	at org.apache.shardingsphere.core.BaseShardingEngine.shard(BaseShardingEngine.java:70)
	at org.apache.shardingsphere.shardingjdbc.jdbc.core.statement.ShardingPreparedStatement.shard(ShardingPreparedStatement.java:224)
	at org.apache.shardingsphere.shardingjdbc.jdbc.core.statement.ShardingPreparedStatement.executeUpdate(ShardingPreparedStatement.java:158)
	at org.jetbrains.exposed.sql.Transaction$exec$2.executeInternal(Transaction.kt:106)
	at org.jetbrains.exposed.sql.statements.Statement.executeIn$exposed(Statement.kt:59)
	at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:183)
	at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:125)
	at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:101)
	at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:92)
	at com.xiachufang.DbUtilsKt.dropTables(DbUtils.kt:22)
	at com.xiachufang.DatasourceKt$main$1.invoke(Datasource.kt:51)
	at com.xiachufang.DatasourceKt$main$1.invoke(Datasource.kt)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$inTopLevelTransaction$1.invoke(ThreadLocalTransactionManager.kt:156)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$inTopLevelTransaction$2.invoke(ThreadLocalTransactionManager.kt:197)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:205)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction(ThreadLocalTransactionManager.kt:196)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$transaction$1.invoke(ThreadLocalTransactionManager.kt:134)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:205)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:106)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:104)
	at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$default(ThreadLocalTransactionManager.kt:103)
	at com.xiachufang.DatasourceKt.main(Datasource.kt:50)
	at com.xiachufang.DatasourceKt.main(Datasource.kt)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.JAXBContext
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 34 more

```

**Comment by terrymanu** [comment]
What is the exception?

**Comment by gfreezy** [comment]
@terrymanu 
Updated in the desc. And more info from 
https://stackoverflow.com/questions/52502189/java-11-package-javax-xml-bind-does-not-exist

**Comment by terrymanu** [comment]
Is there compatible problem if user using  Java7 if add `javax.xml.bind:jaxb-api` dependency?

**Comment by gfreezy** [comment]
Is sharding-jdbc targeted java 7+ ?

**Comment by terrymanu** [comment]
What's mean `how to add a new dependency`?
Is it not just introduce your dependency to maven pom?

**Comment by gfreezy** [comment]
There exists many modules. Should i just add the dependency to all module? Or do we have different dependencies for different modules?

**Comment by gfreezy** [comment]
@terrymanu 
```
./mvnw org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check
```

failed on java 1.7.

```java
[INFO] sharding-integration-test .......................... SKIPPED
[INFO] sharding-jdbc-test 4.0.0-RC3-SNAPSHOT .............. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 6.272 s
[INFO] Finished at: 2019-11-20T02:45:27Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (default-cli) on project shardingsphere: Execution default-cli of goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check failed: Unable to load the mojo 'check' in the plugin 'org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0' due to an API incompatibility: org.codehaus.plexus.component.repository.exception.ComponentLookupException: org/apache/maven/plugins/checkstyle/CheckstyleViolationCheckMojo : Unsupported major.minor version 52.0
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/root/.m2/repository/org/apache/maven/plugins/maven-checkstyle-plugin/3.1.0/maven-checkstyle-plugin-3.1.0.jar
[ERROR] urls[1] = file:/root/.m2/repository/org/sonatype/aether/aether-util/1.7/aether-util-1.7.jar
[ERROR] urls[2] = file:/root/.m2/repository/org/sonatype/sisu/sisu-inject-bean/1.4.2/sisu-inject-bean-1.4.2.jar
[ERROR] urls[3] = file:/root/.m2/repository/org/sonatype/sisu/sisu-guice/2.1.7/sisu-guice-2.1.7-noaop.jar
[ERROR] urls[4] = file:/root/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[ERROR] urls[5] = file:/root/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[6] = file:/root/.m2/repository/org/apache/maven/reporting/maven-reporting-api/3.0/maven-reporting-api-3.0.jar
[ERROR] urls[7] = file:/root/.m2/repository/org/apache/maven/reporting/maven-reporting-impl/2.3/maven-reporting-impl-2.3.jar
[ERROR] urls[8] = file:/root/.m2/repository/org/apache/maven/shared/maven-shared-utils/0.6/maven-shared-utils-0.6.jar
[ERROR] urls[9] = file:/root/.m2/repository/com/google/code/findbugs/jsr305/2.0.1/jsr305-2.0.1.jar
[ERROR] urls[10] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-core/1.2/doxia-core-1.2.jar
[ERROR] urls[11] = file:/root/.m2/repository/xerces/xercesImpl/2.9.1/xercesImpl-2.9.1.jar
[ERROR] urls[12] = file:/root/.m2/repository/xml-apis/xml-apis/1.3.04/xml-apis-1.3.04.jar
[ERROR] urls[13] = file:/root/.m2/repository/org/apache/httpcomponents/httpclient/4.0.2/httpclient-4.0.2.jar
[ERROR] urls[14] = file:/root/.m2/repository/org/apache/httpcomponents/httpcore/4.0.1/httpcore-4.0.1.jar
[ERROR] urls[15] = file:/root/.m2/repository/commons-codec/commons-codec/1.3/commons-codec-1.3.jar
[ERROR] urls[16] = file:/root/.m2/repository/commons-validator/commons-validator/1.3.1/commons-validator-1.3.1.jar
[ERROR] urls[17] = file:/root/.m2/repository/commons-digester/commons-digester/1.6/commons-digester-1.6.jar
[ERROR] urls[18] = file:/root/.m2/repository/commons-logging/commons-logging/1.0.4/commons-logging-1.0.4.jar
[ERROR] urls[19] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-sink-api/1.4/doxia-sink-api-1.4.jar
[ERROR] urls[20] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-logging-api/1.4/doxia-logging-api-1.4.jar
[ERROR] urls[21] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-decoration-model/1.4/doxia-decoration-model-1.4.jar
[ERROR] urls[22] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-site-renderer/1.4/doxia-site-renderer-1.4.jar
[ERROR] urls[23] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-module-xhtml/1.4/doxia-module-xhtml-1.4.jar
[ERROR] urls[24] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-module-fml/1.4/doxia-module-fml-1.4.jar
[ERROR] urls[25] = file:/root/.m2/repository/org/codehaus/plexus/plexus-i18n/1.0-beta-7/plexus-i18n-1.0-beta-7.jar
[ERROR] urls[26] = file:/root/.m2/repository/org/apache/velocity/velocity-tools/2.0/velocity-tools-2.0.jar
[ERROR] urls[27] = file:/root/.m2/repository/commons-chain/commons-chain/1.1/commons-chain-1.1.jar
[ERROR] urls[28] = file:/root/.m2/repository/dom4j/dom4j/1.1/dom4j-1.1.jar
[ERROR] urls[29] = file:/root/.m2/repository/sslext/sslext/1.2-0/sslext-1.2-0.jar
[ERROR] urls[30] = file:/root/.m2/repository/org/apache/struts/struts-core/1.3.8/struts-core-1.3.8.jar
[ERROR] urls[31] = file:/root/.m2/repository/org/apache/struts/struts-taglib/1.3.8/struts-taglib-1.3.8.jar
[ERROR] urls[32] = file:/root/.m2/repository/org/apache/struts/struts-tiles/1.3.8/struts-tiles-1.3.8.jar
[ERROR] urls[33] = file:/root/.m2/repository/org/apache/maven/doxia/doxia-integration-tools/1.6/doxia-integration-tools-1.6.jar
[ERROR] urls[34] = file:/root/.m2/repository/commons-io/commons-io/1.4/commons-io-1.4.jar
[ERROR] urls[35] = file:/root/.m2/repository/junit/junit/3.8.1/junit-3.8.1.jar
[ERROR] urls[36] = file:/root/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.7.1/plexus-component-annotations-1.7.1.jar
[ERROR] urls[37] = file:/root/.m2/repository/org/codehaus/plexus/plexus-resources/1.1.0/plexus-resources-1.1.0.jar
[ERROR] urls[38] = file:/root/.m2/repository/org/codehaus/plexus/plexus-utils/3.0.24/plexus-utils-3.0.24.jar
[ERROR] urls[39] = file:/root/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.24/plexus-interpolation-1.24.jar
[ERROR] urls[40] = file:/root/.m2/repository/org/codehaus/plexus/plexus-velocity/1.1.8/plexus-velocity-1.1.8.jar
[ERROR] urls[41] = file:/root/.m2/repository/com/puppycrawl/tools/checkstyle/8.19/checkstyle-8.19.jar
[ERROR] urls[42] = file:/root/.m2/repository/info/picocli/picocli/3.9.5/picocli-3.9.5.jar
[ERROR] urls[43] = file:/root/.m2/repository/antlr/antlr/2.7.7/antlr-2.7.7.jar
[ERROR] urls[44] = file:/root/.m2/repository/org/antlr/antlr4-runtime/4.7.2/antlr4-runtime-4.7.2.jar
[ERROR] urls[45] = file:/root/.m2/repository/commons-beanutils/commons-beanutils/1.9.3/commons-beanutils-1.9.3.jar
[ERROR] urls[46] = file:/root/.m2/repository/com/google/guava/guava/27.1-jre/guava-27.1-jre.jar
[ERROR] urls[47] = file:/root/.m2/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar
[ERROR] urls[48] = file:/root/.m2/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
[ERROR] urls[49] = file:/root/.m2/repository/org/checkerframework/checker-qual/2.5.2/checker-qual-2.5.2.jar
[ERROR] urls[50] = file:/root/.m2/repository/com/google/errorprone/error_prone_annotations/2.2.0/error_prone_annotations-2.2.0.jar
[ERROR] urls[51] = file:/root/.m2/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar
[ERROR] urls[52] = file:/root/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.17/animal-sniffer-annotations-1.17.jar
[ERROR] urls[53] = file:/root/.m2/repository/net/sf/saxon/Saxon-HE/9.9.1-2/Saxon-HE-9.9.1-2.jar
[ERROR] urls[54] = file:/root/.m2/repository/com/ibm/icu/icu4j/63.1/icu4j-63.1.jar
[ERROR] urls[55] = file:/root/.m2/repository/org/apache/velocity/velocity/1.5/velocity-1.5.jar
[ERROR] urls[56] = file:/root/.m2/repository/commons-lang/commons-lang/2.1/commons-lang-2.1.jar
[ERROR] urls[57] = file:/root/.m2/repository/oro/oro/2.0.8/oro-2.0.8.jar
[ERROR] urls[58] = file:/root/.m2/repository/commons-collections/commons-collections/3.2.1/commons-collections-3.2.1.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[project>org.apache.shardingsphere:shardingsphere:4.0.0-RC3-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]]
[ERROR]
[ERROR] -----------------------------------------------------
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
```

**Comment by terrymanu** [comment]
They are different issue, please just talk about one issue here

**Comment by terrymanu** [comment]
For my understanding, compatible with java7 and java11 is jdk's responsibility, how should we do in our project?

**Comment by gfreezy** [comment]
java7 and java11 is not fully compatible.

**Comment by terrymanu** [comment]
> java7 and java11 are not fully compatible.

But this problem should resolved by JDK, not framework, is it?

**Comment by gfreezy** [comment]
https://www.oracle.com/technetwork/java/javase/11-relnote-issues-5012449.html#JDK-8190378

**Comment by terrymanu** [comment]
We plan upgrade to java8 after 4.0.0 release, does this issue exist in jdk8 too?

**Comment by gfreezy** [comment]
> We plan upgrade to java8 after 4.0.0 release, does this issue exist in jdk8 too?

jdk 8 is fine.

**Comment by terrymanu** [comment]
ok, we can pending this issue until upgrade.

**Comment by kimmking** [comment]
Since jdk9 iwas separated into several jars with the module system.
If you want run a project which built in jdk8/jdk7, we should be modify some config.

**Comment by terrymanu** [comment]
Fixed by #4583
